### PR TITLE
docs: add note to build Angular locally

### PR DESCRIPTION
**Pull request by renntbenrennt**
I got the compiled failed error when I followed the tutorial, which should be informed in advance as there are people who might not know the dependency in the package.json if they have no former relevant experience.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/start/deployment.md [hunk 1]
```diff
@@ -48,6 +48,9 @@ From there you replace the `/src` folder with the one from your `StackBlitz` dow
 ng build --prod
 ```
 
+note: You might also need to append the missing dependencies in your local project's `package.json` file to match the one in the downloaded StackBlitz project before build.
```
Comment (gkalpak) [inline aio/content/start/deployment.md]: But why are the dependencies different?

Comment (SeasonLeee) [reply aio/content/start/deployment.md]: Sorry, as I am still new in Angular and Web dev, I can't find the answer myself. 

These are the dependencies from project I followed the tutorial:

```json
  "dependencies": {
    "@angular/animations": "~8.0.0",
    "@angular/common": "~8.0.0",
    "@angular/compiler": "~8.0.0",
    "@angular/core": "~8.0.0",
    "@angular/forms": "~8.0.0",
    "@angular/platform-browser": "~8.0.0",
    "@angular/platform-browser-dynamic": "~8.0.0",
    "@angular/router": "~8.0.0",
    "rxjs": "~6.4.0",
    "tslib": "^1.9.0",
    "zone.js": "~0.9.1"
  },
```

While these are the dependencies I build on that StackBlitz. online ide:
```json
    "rxjs": "6.5.2",
    "tslib": "1.9.3",
    "core-js": "2.6.9",
    "zone.js": "0.9.1",
    "jasmine-core": "2.99.1",
    "@angular/core": "8.0.0",
    "@angular/forms": "8.0.0",
    "@angular/common": "8.0.0",
    "@angular/router": "8.0.0",
    "jasmine-marbles": "0.5.0",
    "@angular/compiler": "8.0.0",
    "web-animations-js": "2.3.1",
    "@angular/animations": "8.0.0",
    "@angular/platform-browser": "8.0.0",
    "angular-in-memory-web-api": "0.8.0",
    "@angular/platform-browser-dynamic": "8.0.0"
  },
```

The difference is these five:
```
    "jasmine-core": "2.99.1",
    "jasmine-marbles": "0.5.0",
    "core-js": "2.6.9",
    "web-animations-js": "2.3.1",
    "angular-in-memory-web-api": "0.8.0"
```

I submit this pull request is I want to help those who might not notice the difference and have no idea about the error that produce, those who are just like myself new in this technology.

Thanks.

Comment (gkalpak) [reply aio/content/start/deployment.md]: Oh, I see. Yes, we should mention that you need to also update `package.json` dependencies. Thx!
Leaving it to the docs team to review :grin:

Comment (SeasonLeee) [reply aio/content/start/deployment.md]: No problem. 😄 Helping to improve the doc or giving out my advice is the least I can do to give back the community that work unconditionally for the open source work, thank you, guys. And thanks again for reviewed my PR and discussed with me. Thanks.😊

**Review by sreevani-ship-it — COMMENTED** [review]
@SeasonLeee: Please use the following div tag to add notes:

<div class ="alert-is-helpful">

**Note:** You might also need to append the missing dependencies in your local project's `package.json` file to match the one in the downloaded StackBlitz project before the build.

</div>

**Comment by SeasonLeee** [comment]
@chembu Hi, thanks for the review, and sorry I don't see that "following div tag" in your comment. Is that I just use div to wrap the note or should I include some class name in the tag to wrap it? 

I have noticed that there are several class to distinguish tips or other useful thing from the others sentence in the tutorial like `is-helpful`. Should I use this in the div I am about to wrap the note?

Sorry again I can't find the guideline for how to modify documentation...

**Comment by chembu** [comment]
> @chembu Hi, thanks for the review, and sorry I don't see that "following div tag" in your comment. Is that I just use div to wrap the note or should I include some class name in the tag to wrap it?
> 
> I have noticed that there are several class to distinguish tips or other useful thing from the others sentence in the tutorial like `is-helpful`. Should I use this in the div I am about to wrap the note?
> 
> Sorry again I can't find the guideline for how to modify documentation...
Hi,
Please look at the following link for doc-style-guide related to alerts:
https://angular.io/guide/docs-style-guide#alerts

Yes, you need to wrap the note in the div tag with a class name. Hope this helps.

**Comment by SeasonLeee** [comment]
@chembu thanks for sharing the link to the doc guide! I only read the guideline in Angular project README.md which do not elaborate the style for doc, so I guessed there's none. I'm sorry for the inconvenience.

Anyway, I just modified my previous improvement with the div tag using, please review it again, thanks!😊

File: aio/content/start/deployment.md [hunk 2]
```diff
@@ -50,6 +50,12 @@ ng build --prod
 
 This will produce the files that you need to deploy.
 
+<div class="alert is-helpful">
+
+If there are errors about certain packages not found after this command, then you might also need to append the missing dependencies in your local project's `package.json` file to match the one in the downloaded StackBlitz project before build.
```
Comment (chembu) [inline aio/content/start/deployment.md]: Suggestion to rewrite: 

If the above `ng build` command throws an error about missing packages, append the missing dependencies in your local project's `package.json` file to match the one in the downloaded StackBlitz project.

**Review by sreevani-ship-it — COMMENTED** [review]
Suggestion to rewrite:

If the above ng build command throws an error about missing packages, append the missing dependencies in your local project's package.json file to match the one in the downloaded StackBlitz project.

File: aio/content/start/deployment.md (hunk 2 above)
Comment (SeasonLeee) [reply aio/content/start/deployment.md]: Thanks for the suggestion, and I think yours is much better since it's clearer. Thanks again!😊

Comment (chembu) [reply aio/content/start/deployment.md]: Thanks for fixing this @SeasonLeee.

**Comment by chembu** [comment]
LGTM.

File: aio/content/start/deployment.md (hunk 2 above)
Comment (SeasonLeee) [reply aio/content/start/deployment.md]: Not at all!😊 Even I don't have much experience right now, I won't let it stop me from helping others, because I believe helping others is helping myself somehow. And this is the least I can do to give back to the community. 😁 Thanks for all your help, @chembu

File: aio/content/start/deployment.md (hunk 1 above)
Comment (brandonroberts) [reply aio/content/start/deployment.md]: None of those extra dependencies from StackBlitz are required to build the app using the CLI. I think there is a different issue going on here.

Comment (brandonroberts) [reply aio/content/start/deployment.md]: I'd like to understand what the error was so we can correct it if needed instead of saying new users need to guard against errors

Comment (brandonroberts) [reply aio/content/start/deployment.md]: I'd like to understand what the error was so we can correct it if needed instead of saying new users need to guard against errors

Comment (kapunahelewong) [reply aio/content/start/deployment.md]: To what @brandonroberts said here, @SeasonLeee, how did you build the app? With the CLI or another way? The CLI should take care of everything for you so that you have an app up and running quickly without having to edit the `package.json`. If you did use the CLI, mind sharing your version?

Comment (SeasonLeee) [reply aio/content/start/deployment.md]: TLDR: The online tutorial is using Angular 7 CLI and my local's angular is 8 so... this is where the problem is? Maybe?
This is the screenshot of the downloaded file's `package.json`
image

This is the `package.json` from my ng new project
image


@brandonroberts @kapunahelewong   Hi, sorry for the late response, by the time I submitted this issue, I was just in a new company and they are using Angular as main tech stack, but I wasn't so I followed the [official doc](https://angular.io/start) and the build the demo app on [stackblitz.com](https://angular.io/generated/live-examples/getting-started-v0/stackblitz.html)

At the end of this demo thing, I got to deploy section which told me to either download the file from stackblitz.com to build locally or deploy on firebase.

Since I wanted to try locally, I downloaded the file, and unzip it, which just contains 4 things,
image

Then I ng new a new project and copy pasted the file above to replace the related file in the project I just "ng new". And I "ng serve" and I got the errors of "core.js is not found"

Then I look in to the `package.json` and reliaze the package json from my "ng new" is different than the one I download... so... I think this should be addressed... then I opened issue... then... we are here...

Comment (SeasonLeee) [reply aio/content/start/deployment.md]: Well, it's been a while since I opened this issue, by then I was still a newbie in this Angular, although after working with Angular for couple months I still regard myself as one because Angular is so huge and I have a lot to learn. 

Right now this might not be a mistakes I will make but I think it worth to remind those who are not familiar with node or npm to be cautious when the official doc ask people to download the file and build locally.

image

Because, if I follow what the doc suggest, it just failed to compile
image

Comment (kapunahelewong) [reply aio/content/start/deployment.md]: Thanks for these screenshots and the explanation, @SeasonLee. I tried it too and also got an error, but mine is different. @jbogarthyde is going to look into it to see how we might rethink the instructions.

Comment (jbogarthyde) [reply aio/content/start/deployment.md]: I got a similar (but not identical) polyfills error: 
```ERROR in ./src/polyfills.ts
Module not found: Error: Can't resolve 'zone.js/dist/zone-patch-canvas' in 'C:\Users\jbogart\Documents\angular\my-get-started\src'```
When I commented out that line in `polyfills.ts` it built correctly.
@gkalpak Should we be telling people to adjust the polyfills for their dev or deploy systems?

Comment (jbogarthyde) [reply aio/content/start/deployment.md]: When I commented out that line in `polyfills.ts` it built correctly.
@gkalpak Should we be telling people to adjust the polyfills for their dev or deploy systems?

**Comment by jbogarthyde** [comment]
There don't seem to be dependencies missing from `package.json` -- rather, it looks like the `polyfills.ts` file is looking for polyfills that might not be available for the developer's system.

File: aio/content/start/deployment.md (hunk 1 above)
Comment (SeasonLeee) [reply aio/content/start/deployment.md]: @kapunahelewong @jbogarthyde  Hey, thank you all for checking this issue. As my experience grow, it's not that daunting for me to tackle some error messages in the terminal like this. But I really hope the official doc can improve that, because, well, it's official tutorial, right? 😊 Thanks again!😊

**Comment by StephenFluin** [comment]
Here's my summary of the issue. We have a boilerplate package.json that we use for all of the stackblitz examples on the site, when users download the stackblitz app and switch over to a CLI app, the package.json doesn't match and users might not have access to something that was present in stackblitz. The correct long term fix is for the StackBlitz generated project to 100% match and work as a CLI app (cc @EricSimons), but until we have that, the extra text correctly addresses the problem.

**Review by StephenFluin — APPROVED** [review]

**Comment by SeasonLeee** [comment]
@kapunahelewong hey, there 😁 I was so thrilled the other day when I received the message in my email saying that my idea of improving the doc was accepted, (oh my god!!!😆) thanks @StephenFluin for the conclusion! 

But right now, we can see that the merging is blocked!! 

Dued to that inexperienced me in several months ago when I thought different commit messages in the same PR will result in the oldest one.

Screen Shot 2019-10-10 at 11 05 42 PM

That's why I didn't follow the commit message format, SORRY!!!(stupid me...😩)

So, is there another way to fix this? I think the better way to do this is to make a new commit message and make PR again... since from GitHub's doc, which strongly disapprove the means of force push because it might affect other people's local history...

Oh, god, sorry for the trouble again... I was hoping I can help....

But thanks you guys anyway... Now, I'm my way to find some issues that I can handle to tackle~ 

Thanks, again!😆

**Comment by gkalpak** [comment]
FWIW, force-pushing to your branches on your repo (that only exist for the purpose of a PR and only _you_ are likely to be using) it fine :smiley:

**Comment by SeasonLeee** [comment]
@gkalpak hey, thanks for the reply! And that makes perfect sense! Thanks!😆

So now I have `git commit --amend` fixed the latest commit message, but since the branch I have locally is far behind the remote , should I update my local branch with the remote before I force pushing to my repository and make a new PR?

Thanks😁 (so exciting!)

**Comment by kapunahelewong** [comment]
Hi @SeasonLeee! Yes, you'll want to update by rebasing. You might like [my notes of my own workflow](https://gist.github.com/kapunahelewong/567c30f7b353fd1c39bbf6d8343e2dca#how-to-squash). So you'll rebase, which should (if you've squashed already) make your branch be only 1 commit ahead of the remote. Then you push. If you have questions, just let me know. I'll keep a look out for your name in my notifications. 😊

**Comment by SeasonLeee** [comment]
@kapunahelewong Hey, there, thank you so much for the guide and that super awesome note, which is so straightforward and works perfectly!!! OH MY GOD!!! I'm just thrilled for how everything works so smooth by following your note! Thank you😆

Although, questions about what it's actually happen behind that "rebase" and "squash" thing also born in my head while I just follow along the commands, and I have to check out docs and material about them immediately after this comment, thank you for the enlightenment!! 😁

TL;DR:
I have followed the steps in your super awesome notes, and things works in the way it should, I think, because there are no more red cross mark on those automatically test steps, only the last one which I think it means this PR needs another review?
Screen Shot 2019-10-13 at 3 42 53 PM

Last word: 
I hope it all went well so far, and thanks for everyone's help and review and especially the super awesome notes from @kapunahelewong  😁 and taking part in the community is so much fun!!!😆 Thank you!😆

**Comment by jbogarthyde** [comment]
I've fixed the labels, @kapunahelewong should be able to provide the code-owner approval.

**Comment by kapunahelewong** [comment]
@SeasonLeee, thank you so much for your kind words. 🌟 And I'm so happy the steps worked for you! I understand the feeling of not quite getting what's happening with commands, especially when there are a number of them all in a row and none of them seem to be in human language. The way it started making sense to me was just by doing it over and over - and also messing it up a bunch, too! 😅

You've done a great job and really stuck with it! I'll help it get merged. Thank you so much!! ❤️

**Comment by kapunahelewong** [comment]
Caretaker, this is a docs only change. Please let me know if there's anything I can do to help get it merged.

Thanks again, @SeasonLeee!!!

**Review by kapunahelewong — APPROVED** [review]
LGTM!

**Comment by SeasonLeee** [comment]
@kapunahelewong hey, thanks again for everything😁... just one last thing... should I delete this branch? My instinct tells me that I should, since the issue is closed... but it also said that "...branch has unmerged commits", which is why I restored it after I just deleted it... 😅
Screen Shot 2019-10-15 at 1 03 48 PM

Although I can already see my suggestion on the working angular doc site now!!! And it's so surreal!!!! 😆 The minute I realized that I just hopped around telling everyone I know! 😄😅

Again, Engaging in open source super fun and can really learn a lot! (I also tell my friends this😆) 

Thanks again😆

**Comment by santoshyadav198613** [comment]
> @kapunahelewong hey, thanks again for everything... just one last thing... should I delete this branch? My instinct tells me that I should, since the issue is closed... but it also said that "...branch has unmerged commits", which is why I restored it after I just deleted it... 
> Screen Shot 2019-10-15 at 1 03 48 PM
> 
> Although I can already see my suggestion on the working angular doc site now!!! And it's so surreal!!!!  The minute I realized that I just hopped around telling everyone I know! 
> 
> Again, Engaging in open source super fun and can really learn a lot! (I also tell my friends this)
> 
> Thanks again

You can go ahead and delete the branch as this is fork only. This is what I follow, I keep one fork of the master branch and create separate feature branches whenever I want to fix anything.

**Comment by kapunahelewong** [comment]
Yes, @SeasonLeee, you can delete your branch. the unmerged commits message is because of the way PRs are merged (through rebasing rather than through merging, I believe). Good question though. It isn't intuitive.

**Comment by SeasonLeee** [comment]
@kapunahelewong Thank you, thank you, thank you!!!😆 

I hope we will collaborate again on open source project in the future, you have helped a lot!! And it's very important for new people like me in open source world to encounter kind people like you!!! And many thanks to other people who replied this thread! Thanks😆

Thanks!!(I just can't thank you enough😆)

Now I can delete the branch without trembling😅
